src/services/db/__tests__/GitHubService.spec.ts

```diff
@@ -185,11 +281,71 @@ describe("Github Service", () => {
         })
       ).rejects.toThrowError(ConflictError)
       expect(mockAxiosInstance.put).toHaveBeenCalledWith(
-        endpoint,
+        folderEndpoint,
         params,
         authHeader
       )
     })
+
+    it("Create throws an error if parent directory is deleted", async () => {
+      mockAxiosInstance.get.mockImplementation(() => {
+        throw new Error()
+      })
+      await expect(
+        service.create(sessionData, {
+          content,
+          fileName,
+          directoryName,
+        })
+      ).rejects.toThrowError()
+      expect(mockAxiosInstance.get).toHaveBeenCalledWith(folderParentEndpoint, {
+        validateStatus,
+        headers: authHeader.headers,
+        params: {
+          ref: BRANCH_REF,
+        },
+      })
+    })
+
+    it("Create throws an error if a new sub directory is created while the parent directory is deleted", async () => {
+      mockAxiosInstance.get.mockImplementation(() => {
+        throw new Error()
+      })
+      await expect(
+        service.create(sessionData, {
+          content,
+          fileName: subDirectoryFileName,
+          directoryName: subDirectoryName,
+        })
+      ).rejects.toThrowError()
```

Comment:
nit: more specific error
```suggestion
      mockAxiosInstance.get.mockImplementation(() => {
        throw new NotFoundError()
      })
      await expect(
        service.create(sessionData, {
          content,
          fileName: subDirectoryFileName,
          directoryName: subDirectoryName,
        })
      ).rejects.toThrowError(NotFoundError)
```